AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/QtPyScriptCanvasVariableManager.py

```diff
@@ -106,3 +178,29 @@ def validate_variable_count(self, expected) -> None:
         result = helper.wait_for_condition(lambda: expected == row_count, WAIT_TIME_SEC_3)
 
         assert result, "Variable count did not match expected value."
+
+    def toggle_pin_variable_button(self, variable_type: str) -> None:
+        """
+        Function for clicking the pin toggle that appears for variables in the create variable list
+
+        param variable_type: The type of variable you want to pin to the top of the list
+
+        returns None
+        """
+
+        variable_table_view = self.variable_manager.findChild(QtWidgets.QTableView, VARIABLE_PALETTE_QT)
+        variable_object = pyside_utils.find_child_by_pattern(variable_table_view, variable_type)
+
+        toggle_index = 0
+        pinned_toggle = variable_object.siblingAtColumn(toggle_index)
+
+        assert pinned_toggle is not None, "Failed to find variable pinning qt widget"
```

Comment:
Helper opportunity for "find variable pinning qt widget"? then return a tuple of the needed stuff for the rest of the work in the function.

Reply:
abstracted this into two helper functions that use a tuple as suggested. kept the asserts so that data is verified.